lib/frontend/src/frontend.rs

```diff
@@ -52,6 +52,15 @@ struct Position {
     basic_block: Block,
 }
 
+impl Position {
+    fn beginning() -> Self {
```

Comment:
This function name might be a bit dangerous, if it is supposed to imply the entry block of the function. Cretonne doesn't guarantee or require that `ebb0` is the entry block.

IIRC, this was one of those cases where technically there should be an `Option` somewhere because the position briefly needs to be pointing nowhere, but that would require unwraps everywhere.

Cretonne proper has the same problem with the `Layout::entry_block()` function which currently returns an `Option<Ebb>` that is immediately unwrapped at every use.